tsl/src/compression/compression.c

```diff
@@ -2994,11 +2995,18 @@ build_index_scankeys(Relation index_rel, List *index_filters, int *num_scankeys)
 					flags |= SK_SEARCHARRAY;
 				}
 
+				/*
+				 * Check if the filter type is different from the att type. If yes, the
+				 * subtype needs to be set appropriately.
+				 */
+				if (typoid != castNode(Const, filter->value)->consttype)
+					subtype = castNode(Const, filter->value)->consttype;
```

Comment:
Can't we just always include the subtype, even if it is the same?

Reply:
+1 ... if you pass the `consttype` to the `ScanKeyEntryInitialize` you can get rid of this branch and also the variables you've created.

Reply:
@erimatnor @fabriziomello I tried that. But it doesn't work in some cases:

`ERROR:  missing support function 1(1007,1007) in opfamily 1976`